### DISCUSSION TITLE
Some way ala AsyncLocalStorage to retain context inside contributed code

### DISCUSSION THREAD

**Discussion by vaclavHala** [microsoft/vscode-discussions#729] [Extension Development QnA]
I would like to have some way to include session/context token in log messages produced by my extension (think [MDC of java's logback](https://logback.qos.ch/manual/mdc.html), or [span in distributed tracing](https://opentracing.io/docs/overview/spans/)). Often this "session" starts in my code, e.g. command triggered by user, but then as part of its work it will run some vscode APIs like finding/running Tasks. I have tried storing the context token in an [async local store](https://nodejs.org/docs/latest-v16.x/api/async_context.html) which works fine for my parts of the code, but is lost e.g. in `resolveTask` of `TaskProvider`.

Being able to correlate log messages this way is especially valuable in tests when diagnosing failures from failed test run post-mortem.

Simple global `var contextStack = []` would not work as even when the tests run sequentially they might e.g. run multiple tasks in parallel which would corrupt the shared global state.
Is there some mechanism I could use to achieve what I'm trying to do?

**Comment by theReynald** [comment]
Hi vaclavHala. Is it possible to share a link to a screen recording (maybe 'unlisted' on YouTube) of your issue so we can hear you explain the problem? I think due to the depth of your question, a visual demo might make it easier for others to understand your issue to possibly answer.

**Comment by olguzzar** [comment]
Closing this as there's been a period of inactivity and more information is needed. If there are any relevant updates, feel free to reopen!
